File: Assets/Source/Loader.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

namespace Game {
    public class Loader : LuaBehaviour {
        public GameObject gameManager;
        new void Awake() {
        }

        new void Start() {
        }

        public void ManualLuaAwake() {
            base.Awake();
        }
    }
}

File: Assets/Source/lua/Loader.lua
local UE = UnityEngine
local GOInst = UE.GameObject.Instantiate
local GameManager = require"GameManager"
---
-- @module Loader

---
-- @type Loader
-- @extends Game_Loader#Loader
local Loader = {} Loader.__index = Loader

---
-- @function [parent=#Loader] Do
-- @param self
function Loader:Awake()
    if not GameManager.instance then
        UE.GameObject.Instantiate(self.gameManager)
    end
end

return Loader